### PR TITLE
✨  feat: create a new debug tag for docker image with extra tools for hook/lifecycle command usages

### PR DESCRIPTION
**Pull request by mhkarimi1383**
For usages like hook scrips, etc. `kubectl` command is a common thing to have (specially in CI/CD Platforms)

Bitnami's base image has support for both amd64 and arm64 (see https://hub.docker.com/r/bitnami/kubectl/tags) so, it should be OK

### CODE REVIEW

**Comment by zhilyaev** [comment]
1. COPY --chown=root --from=bla-bla would be better.
2. I sure that kubectl is not required inside helmwave image.
   - we can define specific tag or you can build that image on your side.

**Comment by mhkarimi1383** [comment]
@zhilyaev 
I know but kubectl and jq are common commands that I will use in my lifecycle to monitor the condition of CustomResources 
I helmwave supports waiting for conditions (just like how it will do it for normal Workloads when it detects them directly)
I will not need this feature anymore

**Comment by mhkarimi1383** [comment]
@zhilyaev 
I have fixed the COPY Problem.

But `jq` is also needed I think and also my use case is that I want to use `kubectl wait --for=condition=Ready ...` as an example to watch Non-Workload resources that are managing workloads (e.g. for CloudNativePG I have to run `kubectl wait --for=condition=Ready cluster/my-db -n mynamespace` to check if my `Cluster` is up and running)

**Comment by zhilyaev** [comment]
Good job! I will take a look later.

**Comment by mhkarimi1383** [comment]
> Good job! I will take a look later.

Thanks, If it's okay I want to also add jq package too.

**Comment by mhkarimi1383** [comment]
@zhilyaev 
Any updates on this?
I'm installing kubectl on my CI pipeline every time (I don't want to build my own image :])

**Comment by zhilyaev** [comment]
Thank you for PR!
One of the main ideas of helmwave is being simple. And of course, we respect unix-way. That means kubectl binary cannot be included inside the base-image of helmwave. Second point. I would like to save "lego-way" when you can keep your own Dockerfile. I also used this way for a workaround.

**Comment by zhilyaev** [comment]
```Dockerfile
FROM base-release as base-image-kubectl
COPY --chown=root:root --chmod=0775 --from=bitnami/kubectl:latest /opt/bitnami/kubectl/bin/kubectl /bin/kubectl
```

**Comment by mhkarimi1383** [comment]
Thanks
But I have another idea that we can make a debug tag that is fat and contains some extra tools

**Comment by zhilyaev** [comment]
Yeah, It sounds good

**Comment by mhkarimi1383** [comment]
@zhilyaev 
Can you review changes again?
I have refactored my PR to add another tag to container image named `debug`

just like Kaniko, etc.

File: Dockerfile [hunk 1]
```diff
@@ -21,10 +21,21 @@ FROM alpine:${ALPINE_VERSION} as base-release
 RUN apk --update --no-cache add ca-certificates && update-ca-certificates
 ENTRYPOINT ["/bin/helmwave"]
 
+### Base image with shell and debugging tools
+FROM alpine:${ALPINE_VERSION} as base-debug-release
+RUN apk --update --no-cache add ca-certificates jq bash &&\
+	update-ca-certificates
```
Comment (r3nic1e) [inline Dockerfile:27]: ca-certificates are updated automatically during install/upgrade

File: .goreleaser.yml [hunk 2]
```diff
@@ -178,6 +225,23 @@ docker_manifests:
     image_templates:
       - diamon/helmwave:latest-amd64
       - diamon/helmwave:latest-arm64v8
+  - name_template: ghcr.io/helmwave/helmwave:{{ .Version }}-debug
+    image_templates:
+      - ghcr.io/helmwave/helmwave:{{ .Version }}-debug-amd64
+      - ghcr.io/helmwave/helmwave:{{ .Version }}-debug-arm64v8
+  - name_template: ghcr.io/helmwave/helmwave:latest
```
Comment (r3nic1e) [inline .goreleaser.yml:232]: it should be `latest-debug`

**Review by r3nic1e — COMMENTED** [review]
last comments here, will be released in the next release

**Comment by r3nic1e** [comment]
@mhkarimi1383 I am not sure if we want to create debian images right now - it will force us to double number of images which may hit some free ratelimits/quotas.

I guess who wants to make a debian-based image can make it on their own within their registries like this:

```dockerfile
FROM helmwave:version as helmwave

FROM debian:buster-minimal
COPY --from helmwave /usr/local/bin/helmwave /usr/local/bin/helmwave
```

And I don't want to expand os-specific-images approach to include a lot of other distros (ubuntu-based, redhat-based, etc.). Hope that's not a big deal for users

**Comment by r3nic1e** [comment]
hmm, need to do something with qodana and gitleaks actions from forked repos...

File: .goreleaser.yml (hunk 2 above)
Comment (mhkarimi1383) [reply .goreleaser.yml:232]: My first idea was to use `latest-debug`

But I saw other projects using a simple `debug` tag

Comment (r3nic1e) [reply .goreleaser.yml:232]: Okay, so lets drop this one then because you added `debug` tag a bit lower in the code

File: Dockerfile (hunk 1 above)
Comment (mhkarimi1383) [reply Dockerfile:27]: I saw that in older targets :)

File: .goreleaser.yml (hunk 2 above)
Comment (mhkarimi1383) [reply .goreleaser.yml:232]: What about fixed debug tags?

Comment (r3nic1e) [reply .goreleaser.yml:232]: I think it's okay to leave other as is

**Pull request merged**